### PR TITLE
[object-hash] refactor, update version, remove all rules from tslint

### LINKED ISSUE #51598: [object-hash] IOptions is inaccessible.
- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @misak113, @artdiniz

If you do not mention the authors the issue will be ignored.

---

When attempting to use `@types/object-hash`, only the value StaticHash is exported. The result is that `IOptions` is inaccessible and cannot be used as a type in applications. This means that functions that pass values for the options parameter have to use the `any` type, which discards validation.

### PR DESCRIPTION
**Pull request by martin-badin**
Closes: #51598

Documentation: https://github.com/puleos/object-hash

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by EliteMasterEric — APPROVED** [review]
Looks good to me!

**Comment by martin-badin** [comment]
@peterblazejewicz could you review it?

File: types/object-hash/index.d.ts
Comment (peterblazejewicz) [inline types/object-hash/index.d.ts:78]: it should be anything, but not everything:
https://github.com/puleos/object-hash/blob/master/index.js#L23
this fails on unefined:
```js
Uncaught Error: Object argument required.
    at applyDefaults (C:\development\tests\tests-object-hash\node                                                              e_modules\object-hash\index.js:82:11)
```
works with all known types (JS that is), except of undefined, so `{} | null` (or something matching everything except of undefined)

Comment (peterblazejewicz) [reply types/object-hash/index.d.ts:78]: same for everything else in input IMO

**Comment by peterblazejewicz** [comment]
I'd probably use a pattern from adivsed `works everywhere` section:
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html#common-commonjs-patterns
so:
```ts
declare function objectHash(object: {} | null, options?: objectHash.Options): string;

declare namespace objectHash {

    function sha1(object: {} | null): string;
    ...

    interface Stream {
        ...
    }

    interface Options {
        ...
    }
    type BufferEncoding =
        | 'ascii'
        ...
}

export = objectHash;
```
including corrected char case (there is no 'ObjectHash' in original, but there is a function `objectHash` (which stuff added into it.
thoughts?

File: types/object-hash/index.d.ts
Comment (martin-badin) [reply types/object-hash/index.d.ts]: Thanks for your review, updated

**Comment by martin-badin** [comment]
> I'd probably use a pattern from adivsed `works everywhere` section:
> https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html#common-commonjs-patterns
> so:
> 
> ```ts
> declare function objectHash(object: {} | null, options?: objectHash.Options): string;
> 
> declare namespace objectHash {
> 
>     function sha1(object: {} | null): string;
>     ...
> 
>     interface Stream {
>         ...
>     }
> 
>     interface Options {
>         ...
>     }
>     type BufferEncoding =
>         | 'ascii'
>         ...
> }
> 
> export = objectHash;
> ```
> 
> including corrected char case (there is no 'ObjectHash' in original, but there is a function `objectHash` (which stuff added into it.
> thoughts?

Thanks for your awesome review. I updated everything.

**Review by EliteMasterEric — APPROVED** [review]
The tests seem to work so I'm fine with this.

File: types/object-hash/object-hash-tests.ts
Comment (peterblazejewicz) [inline types/object-hash/object-hash-tests.ts:17]: It should work on this one:
```js
hash('')
'67cbf865735ab20856e1e6649eab1c4073bc341e'
```

**Comment by martin-badin** [comment]
@peterblazejewicz thank you again, I updated

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by martin-badin** [comment]
Why can't I merge this MR? I got two approvals.

**Comment by peterblazejewicz** [comment]
IMO we need maintainer ack for this one! Keep calm and keep coding!

**Review by andrewbranch — APPROVED** [review]

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**
